core/src/main/java/org/elasticsearch/action/support/master/TransportMasterNodeAction.java

```diff
@@ -71,6 +77,7 @@ protected TransportMasterNodeAction(Settings settings, String actionName, boolea
         this.transportService = transportService;
         this.clusterService = clusterService;
         this.executor = executor();
+        this.forceLocal = FORCE_LOCAL_SETTING.get(settings);
```

Comment:
rather than introducing a generic setting for this (that people can abuse independently), I would prefer we explicitly check for tribe settings. `TribeService` checks this by using `settings.getGroups("tribe", true))`. We could make `tribe` be a static string and use it here or have  a static method on the `TribeService` called `isTribeNode`.